.travis.yml

```diff
@@ -1,21 +1,85 @@
-env:
-  - GHC_VERSION=7.4.2  CABAL_VERSION=1.18
-  - GHC_VERSION=7.6.3  CABAL_VERSION=1.20
-  - GHC_VERSION=7.8.4  CABAL_VERSION=1.20
-  - GHC_VERSION=7.10.1 CABAL_VERSION=1.22
+# This file has been generated -- see https://github.com/hvr/multi-ghc-travis
+language: c
+sudo: false
+
+cache:
+  directories:
+    - $HOME/.cabsnap
+    - $HOME/.cabal/packages
+
+before_cache:
+  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/build-reports.log
+  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/00-index.tar
+
+matrix:
+  include:
+    - env: CABALVER=1.16 GHCVER=7.4.2
+      compiler: ": #GHC 7.4.2"
+      addons: {apt: {packages: [cabal-install-1.16,ghc-7.4.2], sources: [hvr-ghc]}}
+    - env: CABALVER=1.16 GHCVER=7.6.3
+      compiler: ": #GHC 7.6.3"
+      addons: {apt: {packages: [cabal-install-1.16,ghc-7.6.3], sources: [hvr-ghc]}}
+    - env: CABALVER=1.18 GHCVER=7.8.4
+      compiler: ": #GHC 7.8.4"
+      addons: {apt: {packages: [cabal-install-1.18,ghc-7.8.4], sources: [hvr-ghc]}}
+    - env: CABALVER=1.22 GHCVER=7.10.1
+      compiler: ": #GHC 7.10.1"
+      addons: {apt: {packages: [cabal-install-1.22,ghc-7.10.1], sources: [hvr-ghc]}}
```

Comment:
@CristhianMotoche We can change this to 7.10.2.

Are all four of these taking too long? I guess we don't have to worry about 7.4.2.

In line 73, we can remove the `--enable-benchmarks` I think (not sure if it matters).

Other than these comments, this looks good.